src/pyspiffe/bundle/x509_bundle/x509_bundle_set.py

```diff
@@ -2,41 +2,61 @@
 This module manages X509BundleSet objects.
 """
 
-from typing import Mapping
+from typing import Mapping, List, Optional
+
 from pyspiffe.bundle.x509_bundle.x509_bundle import X509Bundle
 from pyspiffe.spiffe_id.trust_domain import TrustDomain
 
+__all__ = ['X509Bundle']
+
 
 class X509BundleSet(object):
     """X509BundleSet is a set of X509Bundles objects, keyed by trust domain."""
 
-    def __init__(self, bundles: Mapping[TrustDomain, X509Bundle]) -> None:
-        """Creates a new initialized with the given bundles.
+    def __init__(self, bundles_map: Mapping[TrustDomain, X509Bundle]) -> None:
+        """Creates a new initialized Set with the given X509Bundle objects keyed by TrustDomain.
 
         Args:
-            bundles: A set of bundles to initialize the X509BundleSet.
-
+            bundles_map: A map object of X509Bundle objects keyed by TrustDomain to initialize the X509BundleSet.
         """
-        self._bundles = bundles
 
-    def add(self, bundle: X509Bundle):
-        """Adds a new bundle into the set.
+        # create and initialize a dict, so it can be updated (as Mapping doesn't define __setitem__)
+        bundles = {}
+        for pair in bundles_map.items():
+            bundles[pair[0]] = pair[1]
```

Comment:
What do you think about replacing this code by the example below to make it clearer?
```
for td, bundle in bundles_map.items():
    bundles[td] = bundle
```
For each loop iteration, Python will automatically assign the first variable as the key and the second variable as the value for that key.

Reply:
that's indeed a good improvement to my newbie coding style :)